### DISCUSSION TITLE
Using delay or PulseTime

### DISCUSSION THREAD

**Discussion by guydvir2** [arendst/Tasmota#11246] [Q&A] [answered]
Does using a `delay 600` in a `rule` blocks in some way the operation of Tasmota (say for power up , delay and power down), or using `PulseTime` better ?

**Comment by sfromis** [comment, marked as answer]
You probably mean `PulseTime` which is generally a good way, if it works for your case. The `Delay` command is not the best to use, the problem is not it blocking, but instead that it will be cut short by other commands. For rules including programmed timing, you get better results using `RuleTimer`.

**Reply by guydvir2** [reply]
A) Yes, `PulseTime`.
B) The use is for ON timeout ( sort of safety off ).
C) Why `PulseTime` has a 100 offset ?
D) Can you please write an example rule using `RuleTimer` for such case ?

**Comment by meingraham** [comment]
C) For precision on times under one second.
D) Not sure your use case, but take a look at the Rule Cookbook for lots of examples.

**Reply by guydvir2** [reply]
C) TNX
D) Did not find such :/

**Reply by sfromis** [reply]
The rules cookbook has *lots* of examples using RuleTimer

**Reply by guydvir2** [reply]
Right, found it.
`ON button1#state DO Backlog Power1 %value%; RuleTimer1 600 ENDON`
 `ON Rules#Timer=1 DO Power1 off ENDON`

**Comment by meingraham** [comment]
Sorry, I'm not sure what your use case is so unable to understand what your intent is for using a timer.

"Power up" can have a couple of meanings in the Tasmota context. Power up can mean when the device initially boots. And, unfortunately, the term for tuning a relay on and off is referred to as "powering" the relay and uses the `Power` command.

This makes your initial post ambiguous as to what you are trying to achieve (unless I missed some previous context in a different post).

**Reply by guydvir2** [reply]
Sorry for my English. What I meant is `POWER1 1`.
My goal is to create a timeout after `POWER1 1` was initiated.

**Reply by meingraham** [reply]
For what you are trying to do you don't need to overcomplicate it with rules. Set `PulseTime1 700`. Then, when you turn on the relay (e.g., `Power1 1`), Tasmota will turn it off after `PulseTime` for that relay has expired (i.e., 600 seconds).

**Reply by guydvir2** [reply]
You are right :) just wanted to be sure that other tasks are not delayed.
Guy
